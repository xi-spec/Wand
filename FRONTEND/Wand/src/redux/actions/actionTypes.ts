export default {
  LOAD_PRODUCT_TYPE: 'LOAD_PRODUCT_TYPE',
  LOAD_ALL_PRODUCTS: 'LOAD_ALL_PRODUCTS',
  LOAD_NEW_PRODUCTS: 'LOAD_NEW_PRODUCTS',
  LOAD_PRODUCTS_LIST: 'LOAD_PRODUCTS_LIST',
  LOAD_PRODUCT_DETAIL: 'LOAD_PRODUCT_DETAIL',
  ADD_PRODUCT_TO_CART: ' ADD_PRODUCT_TO_CART',
  INCREASE_QUANTITY_CART: 'INCREASE_QUANTITY_CART',
  DECREASE_QUANTITY_CART: 'DECREASE_QUANTITY_CART',
  DELETE_PRODUCT_CART: 'DELETE_PRODUCT_CART',
  DELETE_ALL_CART: 'DELETE_ALL_CART',
  UPDATE_STOCK_API: 'UPDATE_STOCK_API',
  SEARCH_PRODUCT: 'SEARCH_PRODUCT',
  ADD_PRODUCT_TO_FAVOURITE: 'ADD_PRODUCT_TO_FAVOURITE',
  REGISTER: 'REGISTER',
  REGISTER_ERROR: 'REGISTER_ERROR',
  LOGIN: 'LOGIN',
  LOGIN_ERROR: 'LOGIN_ERROR',
  CLEAN_ISEXIST_STATUS: 'CLEAN_STATUS',
  CLEAN_LOGIN_ERROR: 'CLEAN_LOGIN_ERROR',
  CLEAN_REGISTED: 'CLEAN_REGISTED',
  REMOVE_FAVOURITE: 'REMOVE_FAVOURITE',
  LOGOUT: 'LOGOUT',
  UPDATE_USER: 'UPDATE_USER',
  PAYMENT_SUCCSSFULY: 'PAYMENT_SUCCESSFULY',
  PDATE_STOCK_API: 'PDATE_STOCK_API'
}
;
