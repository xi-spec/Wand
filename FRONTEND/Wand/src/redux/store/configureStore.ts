import reduxImmutableStatInvariant from 'redux-immutable-state-invariant';
import thunk from 'redux-thunk';
import { applyMiddleware, compose, createStore } from 'redux';
import rootReducer from '../reducers';

function configureStore () {
  const composeEnhancers = compose;

  return createStore(
    rootReducer,
    composeEnhancers(applyMiddleware(reduxImmutableStatInvariant(), thunk))
  );
}

export default configureStore();
