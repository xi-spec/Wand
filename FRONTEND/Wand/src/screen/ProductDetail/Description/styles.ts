import { StyleSheet } from 'react-native';
import { SIZE, FONTS, COLORS } from '../../../constants/theme';

const styles = StyleSheet.create({
  container: {
    marginHorizontal: 30
  },
  main: {
    marginVertical: 25
  },
  addContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between'

  },
  textColor: {
    letterSpacing: 0.5,
    color: COLORS.detail,
    fontFamily: FONTS.logoRegular,
    fontSize: SIZE.second
  },
  centerContainer: {
    alignItems: 'center'
  },
  stock: {
    textAlign: 'center',
    width: 120,
    backgroundColor: COLORS.green,
    color: 'white',
    borderRadius: 10,
    overflow: 'hidden'
  },
  outStock: {
    textAlign: 'center',
    width: 120,
    backgroundColor: COLORS.second,
    color: 'white',
    borderRadius: 10,
    overflow: 'hidden'

  },
  text: {
    fontFamily: FONTS.primaryMedium,
    fontSize: SIZE.primary
  },
  cost: {
    fontFamily: FONTS.primary,
    fontSize: SIZE.big,
    paddingTop: 15,
    color: COLORS.second
  },
  addButton: {
    width: 120,
    height: 40,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: COLORS.primary,
    marginTop: 10,
    fontFamily: FONTS.primary,
    borderRadius: 5
  },
  disabledButton: {
    width: 120,
    height: 40,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: COLORS.lightGrey,
    borderRadius: 5,
    marginTop: 10
  },
  description: {
    marginTop: 10
  },
  buttonText: {
    color: 'white',
    fontSize: SIZE.primary,
    fontFamily: FONTS.primaryMedium
  },
  title: {
    paddingBottom: 5

  },
  line: {
    width: '100%',
    borderBottomWidth: 1,
    borderBottomColor: COLORS.lightGrey,
    marginVertical: 18
  }
});

export default styles;
