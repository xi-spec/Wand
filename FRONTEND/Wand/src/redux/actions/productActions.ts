import actionTypes from './actionTypes';
import axios from 'axios';
import SERVER from '../../constants/server';
import { Dispatch } from 'redux';
import { Product, ShoppingCart } from '../../models/initiaState';

export function loadProductsType () {
  return async (dispatch : Dispatch) => {
    const { data } = await axios.get(SERVER.URL_TYPE);
    dispatch({
      type: actionTypes.LOAD_PRODUCT_TYPE,
      data
    });
  };
}

export function loadAllProducts () {
  return async (dispatch : Dispatch) => {
    const { data } = await axios.get(SERVER.URL_PRODUCT);
    dispatch({
      type: actionTypes.LOAD_ALL_PRODUCTS,
      data
    });
  };
}

export function loadProductsList (typeName:string) {
  return {
    type: actionTypes.LOAD_PRODUCTS_LIST,
    typeName
  };
}

export function loadProductDetail (productId:string) {
  return {
    type: actionTypes.LOAD_PRODUCT_DETAIL,
    productId
  };
}

export function addProductToCart (product:Product) {
  return {
    type: actionTypes.ADD_PRODUCT_TO_CART,
    product
  };
}

export function increaseQuantityCart (productId:string) {
  return {
    type: actionTypes.INCREASE_QUANTITY_CART,
    productId
  };
}

export function decreaseQuantityCart (productId:string) {
  return {
    type: actionTypes.DECREASE_QUANTITY_CART,
    productId
  };
}

export function deleteProductCart (productId:string, quantity:number) {
  return {
    type: actionTypes.DELETE_PRODUCT_CART,
    productId,
    quantity
  };
}

export function searchProducts (name:string) {
  return {
    type: actionTypes.SEARCH_PRODUCT,
    name
  };
}

export function updateStockApi (cart:ShoppingCart) {
  return (dispatch : Dispatch) => {
    cart.cartList.forEach(async (product) => {
      const { data } = await axios.put(`${SERVER.URL_PRODUCT}/${product._id}`, { stock: product.stock });
      dispatch({
        type: actionTypes.UPDATE_STOCK_API,
        data
      });
    });
  };
}
