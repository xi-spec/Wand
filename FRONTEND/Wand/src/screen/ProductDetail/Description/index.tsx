import React, { useState } from 'react';
import { Text, View, TouchableOpacity } from 'react-native';
import styles from './styles';
import { Product } from '../../../models/initiaState';
import SucesellyAddModal from '../SucesellyAddModal';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { addProductToCart } from '../../../redux/actions/productActions';
import { FAVOURITE_TEXT, CART_TEXT } from '../../../constants/screenTexts';
import setCostValue2Decimals from '../../../utils/setCostValue2Decimals';

function Description ({ selectedProduct, actions, modalVisibleF }:{selectedProduct:Product, actions:any, modalVisibleF:boolean}) {
  const [modalVisible, setModalVisible] = useState(false);

  function showAddModal () {
    setModalVisible(true);
    setTimeout(() => setModalVisible(false), 2000);
  }
  return (
    <View style={styles.container}>
      <SucesellyAddModal modalVisible={modalVisibleF} name={selectedProduct.name} addedText={FAVOURITE_TEXT.ADD_TO_FAVOURITE}/>
      <SucesellyAddModal modalVisible={modalVisible} name={selectedProduct.name} addedText={CART_TEXT.ADD_TO_CART}/>
      <View style={styles.main}>
        <View style={styles.addContainer}>
     <Text style={styles.text}>{selectedProduct.name}</Text>
     <View style={styles.centerContainer}>
     <Text style={selectedProduct.stock ? styles.stock : styles.outStock }>{selectedProduct.stock ? `${selectedProduct.stock} in stock` : 'Out of stock'}</Text>
     </View>

        </View>
     <View style={styles.addContainer}>
     <Text style={styles.cost}>{`${setCostValue2Decimals(selectedProduct.cost.value)} ${selectedProduct.cost.currency}`}</Text>
     <TouchableOpacity
     testID='add'
     disabled={!selectedProduct.stock}
     style={selectedProduct.stock ? styles.addButton : styles.disabledButton}
     onPress={() => { showAddModal(); actions.addProductToCart(selectedProduct); }}>
       <Text style={styles.buttonText}>Add</Text>
     </TouchableOpacity>
     </View>
      </View>
      <View style={styles.description}>
     <Text style={styles.textColor} >{selectedProduct.description}</Text>
     <View style={styles.line}/>
     <Text style={styles.textColor}>{selectedProduct.motto === undefined ? 'What a Nice Day!' : selectedProduct.motto }</Text>
     <View style={styles.line}/>
     <Text style={{ ...styles.text, ...styles.title }}>Product Detail:</Text>
     <Text style={styles.textColor}>{`Materials: ${selectedProduct.material}`}</Text>
     <Text style={styles.textColor}>{`Measurements: ${selectedProduct.measurements}`}</Text>
     <Text style={styles.textColor}>{`Weight: ${selectedProduct.weight.value}${selectedProduct.weight.volum}`}</Text>
     <Text style={styles.textColor}>{`Packaging: ${selectedProduct.Packaging}`}</Text>
     <View style={styles.line}/>
      </View>
    </View>
  );
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators({
      addProductToCart
    }, dispatch)
  };
}

export default connect(null, mapDispatchToProps)(Description)
;
