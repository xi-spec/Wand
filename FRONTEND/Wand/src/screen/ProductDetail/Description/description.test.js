import React from 'react';
import Description from './index';
import configureStore from 'redux-mock-store';
import * as actions from '../../../redux/actions/productActions';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { render, fireEvent, act } from '@testing-library/react-native';

jest.mock('../../../redux/actions/productActions');

describe('Given a Description component', () => {
  const mockStore = configureStore([thunk]);
  const store = mockStore({

  });
  let selectedProduct;

  beforeEach(() => {
    jest.spyOn(actions, 'addProductToCart').mockReturnValueOnce({ type: '' });
    selectedProduct = {
      name: 'rings',
      images: { alls: ['url'] },
      cost: {
        value: 2,
        currency: 'euro'
      },
      weight: {
        value: 2,
        volum: 'g'
      },
      stock: 1
    };
  });

  describe('When is rendered with selectedProduct.stock', () => {
    test('Then is rendered correct', () => {
      const tree = render(<Provider store={store}><Description selectedProduct={selectedProduct} /></Provider>);
      expect(tree).toMatchSnapshot();
    });
  });
  describe('When is rendered without selectedProduct.stock', () => {
    test('Then is rendered correct', () => {
      selectedProduct.stock = undefined;
      const tree = render(<Provider store={store}><Description selectedProduct={selectedProduct} /></Provider>);
      expect(tree).toMatchSnapshot();
    });
  });

  describe('When is rendered with the selectedProduct.motto = hello', () => {
    test('Then is rendered correct', () => {
      selectedProduct.motto = 'hello';
      const tree = render(<Provider store={store}><Description selectedProduct={selectedProduct} /></Provider>);
      expect(tree).toMatchSnapshot();
    });
  });

  describe('When the add button is pressed', () => {
    test('Then actions.addProductToCart is invoked', async () => {
      const { getByTestId } = render(<Provider store={store}><Description selectedProduct={selectedProduct} /></Provider>);
      const add = getByTestId('add');
      act(() => {
        fireEvent.press(add);
      });
      expect(actions.addProductToCart).toHaveBeenCalled();
    });
    test('Then Description component is to match snapshoot', () => {
      jest.useFakeTimers();
      const tree = render(<Provider store={store}><Description selectedProduct={selectedProduct} /></Provider>);
      const add = tree.getByTestId('add');

      act(() => {
        fireEvent.press(add);
        jest.runAllTimers();
      });
      expect(tree).toMatchSnapshot();
    });
  });
});
