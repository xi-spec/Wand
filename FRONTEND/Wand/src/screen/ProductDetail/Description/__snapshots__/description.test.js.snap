// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Description component When is rendered with selectedProduct.stock Then is rendered correct 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 30,
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F3603F",
          "paddingTop": 60,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "paddingBottom": 20,
          }
        }
      >
        "rings" has been added to your wishlist.
      </Text>
    </View>
  </Modal>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F3603F",
          "paddingTop": 60,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "paddingBottom": 20,
          }
        }
      >
        "rings" has been added to your cart.
      </Text>
    </View>
  </Modal>
  <View
    style={
      Object {
        "marginVertical": 25,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "MontserratMedium",
            "fontSize": 18,
          }
        }
      >
        rings
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "#63B798",
              "borderRadius": 10,
              "color": "white",
              "overflow": "hidden",
              "textAlign": "center",
              "width": 120,
            }
          }
        >
          1 in stock
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#F3603F",
            "fontFamily": "MontserratRegular",
            "fontSize": 22,
            "paddingTop": 15,
          }
        }
      >
        2.00 euro
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2E3192",
            "borderRadius": 5,
            "fontFamily": "MontserratRegular",
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "opacity": 1,
            "width": 120,
          }
        }
        testID="add"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "MontserratMedium",
              "fontSize": 18,
            }
          }
        >
          Add
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    />
    <View
      style={
        Object {
          "borderBottomColor": "#E6E6E6",
          "borderBottomWidth": 1,
          "marginVertical": 18,
          "width": "100%",
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      What a Nice Day!
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#E6E6E6",
          "borderBottomWidth": 1,
          "marginVertical": 18,
          "width": "100%",
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "MontserratMedium",
          "fontSize": 18,
          "paddingBottom": 5,
        }
      }
    >
      Product Detail:
    </Text>
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      Materials: undefined
    </Text>
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      Measurements: undefined
    </Text>
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      Weight: 2g
    </Text>
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      Packaging: undefined
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#E6E6E6",
          "borderBottomWidth": 1,
          "marginVertical": 18,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;

exports[`Given a Description component When is rendered with the selectedProduct.motto = hello Then is rendered correct 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 30,
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F3603F",
          "paddingTop": 60,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "paddingBottom": 20,
          }
        }
      >
        "rings" has been added to your wishlist.
      </Text>
    </View>
  </Modal>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F3603F",
          "paddingTop": 60,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "paddingBottom": 20,
          }
        }
      >
        "rings" has been added to your cart.
      </Text>
    </View>
  </Modal>
  <View
    style={
      Object {
        "marginVertical": 25,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "MontserratMedium",
            "fontSize": 18,
          }
        }
      >
        rings
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "#63B798",
              "borderRadius": 10,
              "color": "white",
              "overflow": "hidden",
              "textAlign": "center",
              "width": 120,
            }
          }
        >
          1 in stock
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#F3603F",
            "fontFamily": "MontserratRegular",
            "fontSize": 22,
            "paddingTop": 15,
          }
        }
      >
        2.00 euro
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2E3192",
            "borderRadius": 5,
            "fontFamily": "MontserratRegular",
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "opacity": 1,
            "width": 120,
          }
        }
        testID="add"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "MontserratMedium",
              "fontSize": 18,
            }
          }
        >
          Add
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    />
    <View
      style={
        Object {
          "borderBottomColor": "#E6E6E6",
          "borderBottomWidth": 1,
          "marginVertical": 18,
          "width": "100%",
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      hello
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#E6E6E6",
          "borderBottomWidth": 1,
          "marginVertical": 18,
          "width": "100%",
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "MontserratMedium",
          "fontSize": 18,
          "paddingBottom": 5,
        }
      }
    >
      Product Detail:
    </Text>
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      Materials: undefined
    </Text>
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      Measurements: undefined
    </Text>
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      Weight: 2g
    </Text>
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      Packaging: undefined
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#E6E6E6",
          "borderBottomWidth": 1,
          "marginVertical": 18,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;

exports[`Given a Description component When is rendered without selectedProduct.stock Then is rendered correct 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 30,
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F3603F",
          "paddingTop": 60,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "paddingBottom": 20,
          }
        }
      >
        "rings" has been added to your wishlist.
      </Text>
    </View>
  </Modal>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F3603F",
          "paddingTop": 60,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "paddingBottom": 20,
          }
        }
      >
        "rings" has been added to your cart.
      </Text>
    </View>
  </Modal>
  <View
    style={
      Object {
        "marginVertical": 25,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "MontserratMedium",
            "fontSize": 18,
          }
        }
      >
        rings
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "#F3603F",
              "borderRadius": 10,
              "color": "white",
              "overflow": "hidden",
              "textAlign": "center",
              "width": 120,
            }
          }
        >
          Out of stock
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#F3603F",
            "fontFamily": "MontserratRegular",
            "fontSize": 22,
            "paddingTop": 15,
          }
        }
      >
        2.00 euro
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#E6E6E6",
            "borderRadius": 5,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "opacity": 1,
            "width": 120,
          }
        }
        testID="add"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "MontserratMedium",
              "fontSize": 18,
            }
          }
        >
          Add
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    />
    <View
      style={
        Object {
          "borderBottomColor": "#E6E6E6",
          "borderBottomWidth": 1,
          "marginVertical": 18,
          "width": "100%",
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      What a Nice Day!
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#E6E6E6",
          "borderBottomWidth": 1,
          "marginVertical": 18,
          "width": "100%",
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "MontserratMedium",
          "fontSize": 18,
          "paddingBottom": 5,
        }
      }
    >
      Product Detail:
    </Text>
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      Materials: undefined
    </Text>
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      Measurements: undefined
    </Text>
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      Weight: 2g
    </Text>
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      Packaging: undefined
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#E6E6E6",
          "borderBottomWidth": 1,
          "marginVertical": 18,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;

exports[`Given a Description component When the add button is pressed Then Description component is to match snapshoot 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 30,
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F3603F",
          "paddingTop": 60,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "paddingBottom": 20,
          }
        }
      >
        "rings" has been added to your wishlist.
      </Text>
    </View>
  </Modal>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F3603F",
          "paddingTop": 60,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "paddingBottom": 20,
          }
        }
      >
        "rings" has been added to your cart.
      </Text>
    </View>
  </Modal>
  <View
    style={
      Object {
        "marginVertical": 25,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "MontserratMedium",
            "fontSize": 18,
          }
        }
      >
        rings
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "#63B798",
              "borderRadius": 10,
              "color": "white",
              "overflow": "hidden",
              "textAlign": "center",
              "width": 120,
            }
          }
        >
          1 in stock
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#F3603F",
            "fontFamily": "MontserratRegular",
            "fontSize": 22,
            "paddingTop": 15,
          }
        }
      >
        2.00 euro
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2E3192",
            "borderRadius": 5,
            "fontFamily": "MontserratRegular",
            "height": 40,
            "justifyContent": "center",
            "marginTop": 10,
            "opacity": 1,
            "width": 120,
          }
        }
        testID="add"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "MontserratMedium",
              "fontSize": 18,
            }
          }
        >
          Add
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    />
    <View
      style={
        Object {
          "borderBottomColor": "#E6E6E6",
          "borderBottomWidth": 1,
          "marginVertical": 18,
          "width": "100%",
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      What a Nice Day!
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#E6E6E6",
          "borderBottomWidth": 1,
          "marginVertical": 18,
          "width": "100%",
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "MontserratMedium",
          "fontSize": 18,
          "paddingBottom": 5,
        }
      }
    >
      Product Detail:
    </Text>
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      Materials: undefined
    </Text>
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      Measurements: undefined
    </Text>
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      Weight: 2g
    </Text>
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "RalewayRegular",
          "fontSize": 14,
          "letterSpacing": 0.5,
        }
      }
    >
      Packaging: undefined
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#E6E6E6",
          "borderBottomWidth": 1,
          "marginVertical": 18,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;
